File: intro-prog/lista 6/ex11.py
"""
11. Escreva uma função que recebe uma lista vet e devolve o maior e o menor elementos. Obs: Não é
permitido o uso de mint() e max()
"""

def findMinMax(ls):
    ls.sort()
    return ls[0], ls[len(ls) - 1]

def main():
    print(f"{findMinMax([8, 1, 23, 10, 4])}")

main()

File: Algoritmos/Lista 2/ex11.c
/*11. Escreva uma função que recebe um vetor vet de tamanho n e inverte os seus elementos (no
próprio vetor*/

#include <stdio.h>

void reverse(int n, int v[]){
    int aux = 0;

    for(int i = 0; i < n/2; i++){
        aux = v[i];
        v[i] = v[n-1-i];
        v[n-1-i] = aux;
    }
}

void printArray(int n, int v[]){
    for(int i = 0; i < n; i++){
        printf("%d ", v[i]);
    }
}

int main(){
    int v[] = {1, 2, 3, 4, 5};
    reverse(5, v);
    printArray(5, v);
}